src/index.js

```diff
@@ -3,40 +3,118 @@
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
 
-  //... your code goes here
+  const price = product.querySelector(".price span").innerHTML
+  const quantity = product.querySelector(".quantity input").value
+
+  let newProduct = price * quantity
+  console.log(newProduct)
+  product.querySelector(".subtotal span").innerText = newProduct
+  return newProduct
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+  //const singleProduct = document.querySelector('.product');
+  //updateSubtotal(singleProduct);
   // end of test
 
   // ITERATION 2
-  //... your code goes here
+  const multipleProducts = document.querySelectorAll(".product")
+  console.log(multipleProducts)
+
+  let total = 0
+
+  multipleProducts.forEach(product => { 
+
+    total += updateSubtotal(product)
+    console.log(total)
+
+  })
 
   // ITERATION 3
-  //... your code goes here
+
+  document.querySelector("#total-value span").innerHTML = total
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
-  const target = event.currentTarget;
+  //console.log(event.currentTarget.parentNode.previousSibling.previousSibling.lastChild.innerText)
+
+  document.querySelector("#total-value span").innerHTML -= event.currentTarget.parentNode.previousSibling.previousSibling.lastChild.innerText
+  
+  const target = event.currentTarget.parentNode.parentNode;
   console.log('The target in remove is:', target);
-  //... your code goes here
+
+  target.remove()
 }
 
 // ITERATION 5
 
 function createProduct() {
```

Comment:
For this function, instead of slowly adding classes and child nodes to your tablerow, you could perhaps write your HTML structure as a string and store it in a variable, before appending it to your list. Still works, and that's what matters!